### PR TITLE
python3Packages.hwi: 1.2.1 -> 2.0.0

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

new major release

supersedes https://github.com/NixOS/nixpkgs/pull/116592

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116598` at 76df5842 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build (new failures):</summary>
  <ul>
    <li><a href="https://gist.github.com/dc57ed1b284ec03f69195701f5a4490e">hwi</a></li> (python38Packages.hwi)</li>
    <li><a href="https://gist.github.com/3b92d1690dd3b32a621384590acc4b7b">python39Packages.hwi</a></li></li>
  </ul>
</details>

**Comment by prusnak** [comment]
@GrahamcOfBorg build python3Packages.hwi

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 116598` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>hwi (python38Packages.hwi)</li>
    <li>python39Packages.hwi</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package built (see https://github.com/NixOS/nixpkgs/pull/116598#issuecomment-801112297)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-116598]$ ./results/hwi/bin/hwi -h
usage: hwi [-h] [--device-path DEVICE_PATH] [--device-type DEVICE_TYPE] [--password PASSWORD] [--stdinpass] [--chain {main,test,regtest,signet}] [--debug] [--fingerprint FINGERPRINT] [--version] [--stdin] [--interactive] [--expert]
           {enumerate,getmasterxpub,signtx,getxpub,signmessage,getkeypool,getdescriptors,displayaddress,setup,wipe,restore,backup,promptpin,togglepassphrase,sendpin,installudevrules} ...

Hardware Wallet Interface, version 2.0.0.
Access and send commands to a hardware wallet device. Responses are in JSON format.
[...]
ew/pr-116598/results/python39Packages.hwi/lib/python3.9/site-packages]$ python
Python 3.8.8 (default, Feb 19 2021, 11:04:50) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from hwilib import hwwclient
>>> device = hwwclient.HardwareWalletClient(path, password, expert)
>>> device = hwwclient.HardwareWalletClient("", "", True)      
>>> device.fingerprint
>>> print(device.fingerprint)
None
```

Update looks good to me.

File: pkgs/development/python-modules/hwi/default.nix [hunk 1]
```diff
@@ -7,35 +7,28 @@
...
-  '';
-
   propagatedBuildInputs = [
     bitbox02
     ecdsa
     hidapi
     libusb1
     mnemonic
     pyaes
+    typing-extensions
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hwi/default.nix:31]: This package shouldn't be required in 3.7.

**Pull request merged**